pkg/sql/plan/query_builder.go

```diff
@@ -433,15 +434,15 @@ func (builder *QueryBuilder) remapAllColRefs(nodeID int32, colRefCnt map[[2]int3
 				Expr: &plan.Expr_Col{
 					Col: &ColRef{
 						RelPos: -2,
-						ColPos: int32(idx),
+						ColPos: int32(idx) + groupSize,
```

Comment:
Why you modify this? The `ColPos` of aggregates are starting from 0.

Reply:
ColPos will plus groupSize during compile, but it will change the cached plan. So do this during buildPlan and delete the related logic in compile.

Reply:
Compile should not change the cached plan. That's a bad design. The plan should be cached before any temporary modification in compilation.